lib/AST/Expr.cpp

```diff
@@ -1456,6 +1456,13 @@ DestructureTupleExpr::create(ASTContext &ctx,
                                          srcExpr, dstExpr, ty);
 }
 
+FunctionConversionExpr::FunctionConversionExpr(Expr *subExpr, Type type)
+    : ImplicitConversionExpr(ExprKind::FunctionConversion, subExpr, type) {
+  if (auto *CE = dyn_cast<ClosureExpr>(subExpr)) {
```

Comment:
Any chance this could get tripped up by an `ImplicitCastExpr` or `ParenExpr` surrounding the `ClosureExpr`?